blivet/formats/fs.py

```diff
@@ -436,12 +439,27 @@ def _post_create(self, **kwargs):
             self.write_uuid()
 
     def _pre_resize(self):
-        # file systems need a check before being resized
-        self.do_check()
+        if self.status:
+            if flags.allow_online_fs_resize:
+                if self.target_size > self.size and not self._resize_support & FSResize.ONLINE_GROW:
+                    raise FSError("This filesystem doesn't support online growing")
+                if self.target_size < self.size and not self._resize_support & FSResize.ONLINE_SHRINK:
+                    raise FSError("This filesystem doesn't support online shrinking")
+            else:
+                raise FSError("Resizing of mounted filesystems is disabled")
```

Comment:
In general should this be considered an error or do we want to try to unmount the filesystem right now? Currently for partitions and LVs the block device `_pre_resize` functions unmount (`teardown`) the format if the `allow_online_fs_resize` flag isn't set, but we might try to unmount the format and do the online resize only when the unmount fails.